hugr-py/src/hugr/serialization/tys.py

```diff
@@ -475,9 +480,10 @@ def deserialize(self) -> tys.Type:
 
 def model_rebuild(
     classes: Mapping[str, type],
-    config: ConfigDict = ConfigDict(),
+    config: ConfigDict | None = None,
     **kwargs,
 ):
+    config = config or ConfigDict()
```

Comment:
This seems worse?

Reply:
Looks like it's this lint: https://docs.astral.sh/ruff/rules/function-call-in-default-argument/
It seems we can add exceptions, but I can live with this convention

Reply:
python generally doesn't like functions being called in the arguments, taking none then calling the default function inside the body is the preferred pattern

https://docs.astral.sh/ruff/rules/function-call-in-default-argument/

Reply:
By convention, I'd expect nullary constructors like `MyStruct()` to be benign and more acceptable to put here than an arbitrary function. Maybe too much c++ brain...